src/org/omegat/core/machinetranslators/ApertiumTranslate.java

```diff
@@ -43,9 +51,12 @@
  * @author Didier Briel
  */
 public class ApertiumTranslate extends BaseTranslate {
-    protected static final String GT_URL = "https://www.apertium.org/apy/translate?q=";
+    protected static final String PROPERTY_APERTIUM_CUSTOM = "apertium.server.custom";
+    protected static final String PROPERTY_APERTIUM_URL = "/translate?q=";
+    protected static final String PROPERTY_APERTIUM_URL2 = "&markUnknown=no&langpair=#sourceLang#|#targetLang#&key=";
+    protected static final String PROPERTY_APERTIUM_SERVER = "https://www.apertium.org/apy";
     // Specific OmegaT key
-    protected static final String GT_URL2 = "&markUnknown=no&langpair=#sourceLang#|#targetLang#&key=bwuxb5jS+VwSJ8mLz1qMfmMrDGA";
+    protected static final String PROPERTY_APERTIUM_API_KEY = "bwuxb5jS+VwSJ8mLz1qMfmMrDGA";
```

Comment:
The value of a `PROPERTY_*` constant is meant to be a human-readable key used for storing the value in preferences and for supplying the value on the command line. See e.g. the [Microsoft MT connector](https://github.com/omegat-org/omegat/blob/3dff23f3af974d1a1b8015375255b60bb91e6879/src/org/omegat/core/machinetranslators/MicrosoftTranslatorAzure.java#L61)

Here they are doing double duty, serving as a (human-*un*readable) keys *and* the actual values when no other value has been supplied.

Please make separate definitions for keys and default values.

Reply:
Thanks, I agree that the approach was messy. Fixed in latest commit.